src/main/php/PDepend/Metrics/AbstractCachingAnalyzer.php

```diff
@@ -56,6 +56,8 @@
  * This abstract class provides an analyzer that provides the basic infrastructure
  * for caching.
  *
+ * @template T of numeric|array<string, numeric>
```

Comment:
Rather then limit it to this I think it would be better to allow any value, this will make it more flexible for anyone using PDepend to build on top of. And still give us full evaluation since you made it a class template value.

```suggestion
 * @template T
```